### PR TITLE
Issue 79: Make some functions internal

### LINKED ISSUE #79: Decide on which functions to make internal
Currently (approximately) all of the functions in the package are external (`@export`).

We should decide which should be made internal to the package.

As example, in #70 @seabbs raised this regarding functions in `utils.R` and the `epidist_stancode` functions.

### PR DESCRIPTION
**Pull request by athowes**
## Description

This draft PR closes #79.

It is my proposal for which functions to be made internal:

* Any `utils.R`
* Any custom `brms` family prediction and log likelihood functions
* I also moved `replace_prior` into `utils.R`

I used `#' @noRd`. Another alternative would be to use `#' @keywords internal`?

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [ ] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [ ] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by seabbs** [comment]
> #' @keywords internal

I would prefer this

File: R/utils.R [hunk 1]
```diff
@@ -39,8 +37,53 @@ epidist_version_stanvar <- function() {
...
+#' This function takes `old_prior` and replaces any prior distributions
+#' contained in it by the corresponding prior distribution in `new_prior`.
+#' If there is a prior distribution in `new_prior` with no match in `old_prior`
+#' then the function will error and give the name of the new prior distribution
+#' with no match.
+#'
+#' @param old_prior One or more prior distributions in the class `brmsprior`
+#' @param new_prior One or more prior distributions in the class `brmsprior`
+#' @importFrom cli cli_inform
+#' @importFrom utils capture.output
```
Comment (seabbs) [inline R/utils.R:57]: yourt missing the dplyr functions from importFrom

**Comment by athowes** [comment]
> I would prefer this

Why?

File: R/utils.R [hunk 2]
```diff
@@ -6,8 +6,7 @@
 #' @param path The path within the `stan/` folder of the installed `epidist`
 #' package to the Stan code chunk of interest.
 #' @return A character string containing the Stan code chunk of interest.
-#' @family utils
-#' @export
+#' @noRd
```
Comment (seabbs) [inline R/utils.R:9]: for all internal functions we might consider a rename. 1. They likely don't need the epidist prefix and 2. we might want to use common these are internal signals like . or _ as a prefix

**Review by seabbs — COMMENTED** [review]
I agree with all the suggestions to go internal. Some specific points about the process.

**Comment by seabbs** [comment]
> Why?

because then they are still doc'd if people want to know how they work. I think that the docs for noRd have a warning i.e do you really want to do this as I think the default is that they are internal but still have a help fie

File: R/utils.R (hunk 2 above)
Comment (athowes) [reply R/utils.R:9]: Happy with remove `epidist_`. Don't have major desire to do `.` or `_` but if there is a push I can do it

Comment (seabbs) [reply R/utils.R:9]: The reason is to indicate these are internal functions vs not. Its generally pretty standard practice so I would favour doing so but I don't feel strongly

Comment (athowes) [reply R/utils.R:9]: For the `brms` ones I think that would break things

Comment (athowes) [reply R/utils.R]: [a3447be](https://github.com/epinowcast/epidist/pull/300/commits/a3447be829f7a91d62270020d08925bdbb1c7821) does the remove `epidist_` part

File: R/utils.R (hunk 1 above)
Comment (athowes) [reply R/utils.R:57]: [19104b3](https://github.com/epinowcast/epidist/pull/300/commits/19104b3f08cce359d07c2811815ed17bc8b2bd15)

File: R/utils.R (hunk 2 above)
Comment (seabbs) [reply R/utils.R]: > For the brms ones I think that would break things

Then don't do it for those ones. What is the push back for the others?

Comment (athowes) [reply R/utils.R]: It's just not something I'm familiar with standard practise on / I have some doubts about the extent to which it is standard practise. But OK as I say I don't mind so I'll add it

Comment (athowes) [reply R/utils.R]: "Have some doubts" --> pick an R package I think they would know what they are doing, then what proportion of the time are they doing this? For example I picked `ggplot2` and I don't see it there

Comment (pearsonca) [reply R/utils.R]: can't start with underscores - that's just a weird bit of R naming rules (https://cran.r-project.org/doc/FAQ/R-FAQ.html#What-are-valid-names_003f). I've definitely seen the starts-with-dot for internals.

Comment (pearsonca) [reply R/utils.R]: (it's probably leaning into the base R `.Internal()` as an emphasis idiom)

Comment (athowes) [reply R/utils.R]: I think done in [d45d0c9](https://github.com/epinowcast/epidist/pull/300/commits/d45d0c9afdd08711badba3e6212d6680706c05ab)

**Comment by athowes** [comment]
Yeah probably, is there a good way to do this rebase? When I do `pull origin main` on the branch then I have all these docs conflicts but seems bad to be going through and hand fixing them. Want to pull the non-docs then run `document`

**Comment by seabbs** [comment]
> Edit: I tried the second option, keep all conflicts then overwrite.

Sorry yes this is what I would do

**Review by seabbs — APPROVED** [review]

**Pull request merged**